### PR TITLE
Adjust nests value by evolution

### PR DESCRIPTION
**Pull request by RedRafe**
Biter nests health increases with evo (see [FFF-427](https://factorio.com/blog/post/fff-427)). This adjusts the score obtained by killing spawners according to the new approximated function of:

`y = 350 + 3150 * (x^2.25)`

resulting in

`factor = 1 + 9 * (evolution^2.25)`

which describes the factor of `nest_health / base_prototype_health` in function of evo(x). Note that this only applies up to 100% evo, after that, the threat reduction for killing (and health value of) the biter nests does not change.

graph

### CODE REVIEW

File: maps/biter_battles_v2/ai.lua [hunk 1]
```diff
@@ -449,6 +449,10 @@ function Public.subtract_threat(entity)
         local health_buff_equivalent_revive = storage.biter_health_factor[game.forces[biter_not_boss_force].index]
         factor = bb_config.health_multiplier_boss * health_buff_equivalent_revive
     end
+    if entity.type == 'unit-spawner' then
```
Comment (clifffrey) [inline maps/biter_battles_v2/ai.lua:452]: I would use `elseif` or I would change it to be `factor = factor * ...`

File: maps/biter_battles_v2/functions.lua [hunk 2]
```diff
@@ -657,4 +657,19 @@ function Functions.clear_corpses(player, param)
     player.print('Cleared biter-corpses.', { color = Color.success })
 end
 
+--- See the docs for LuaPlayer::create_local_flying_text, + surface param
+---@param params table
```
Comment (clifffrey) [inline maps/biter_battles_v2/functions.lua:661]: you can make the type checker happier (or at least pickier) if you do this as
```
---@param params {surface: string|integer|LuaSurface}
```

**Review by clifffrey — CHANGES_REQUESTED** [review]
It would be good to update the summary to be explicit that this only increases the threat-benefit of killing nests up to 100% evolution.  Specifically that the health/threat increases do not continue to scale after that.

I feel like this will be a net-buff to threatfarming, but I could be wrong.  So many things about the balance might need tweaking post 2.0, and this change seems like a good start/approximation of what might be good, so I like it.

File: maps/biter_battles_v2/ai.lua (hunk 1 above)
Comment (RedRafe) [reply maps/biter_battles_v2/ai.lua:452]: done

File: maps/biter_battles_v2/functions.lua (hunk 2 above)
Comment (RedRafe) [reply maps/biter_battles_v2/functions.lua:661]: ah, thanks!

**Comment by RedRafe** [comment]
@clifffrey sorry, which summary?

**Comment by clifffrey** [comment]
> @clifffrey sorry, which summary?

I just meant the PR description, which will turn in to the commit description.
I edited it, adding "Note that this only applies up to 100% evo, after that, the threat reduction for killing (and health value of) the biter nests does not change."

**Review by clifffrey — APPROVED** [review]

**Pull request merged**